src/audio/kpb.c

```diff
@@ -126,6 +136,51 @@ static uint64_t kpb_task_deadline(void *data)
 	return SOF_TASK_DEADLINE_ALMOST_IDLE;
 }
 
+#if CONFIG_AMS
+void kpb_on_ams_kpd_notification(const struct ams_message_payload *const ams_message_payload,
+				 void *ctx)
+{
+	struct kpb_client *cli_data = (struct kpb_client *)ams_message_payload->message;
+	struct comp_dev *dev = ctx;
+
+	comp_dbg(dev, "kpb_on_ams_kpd_notification()");
+
+	kpb_init_draining(dev, cli_data);
+}
+
+static int kpb_register_ams_consumer(struct comp_dev *dev)
+{
+	struct comp_data *kpb = comp_get_drvdata(dev);
+	const uint8_t *ams_kpd_msg = (const uint8_t *)SOF_RT_UUID(ams_kpd_msg_uuid);
+	uint16_t mod_id, inst_id;
+	int ret;
+
+	comp_dbg(dev, "kpb_register_ams_consumer()");
+
+	mod_id = IPC4_MOD_ID(dev_comp_id(dev));
+	inst_id = IPC4_INST_ID(dev_comp_id(dev));
+
+	ret = ams_get_message_type_id(ams_kpd_msg, &kpb->am_uuid_id);
+	if (ret)
+		return ret;
+
+	return ams_register_consumer(kpb->am_uuid_id, mod_id, inst_id,
+				     kpb_on_ams_kpd_notification, dev);
+}
+
+static int kpb_unregister_ams_consumer(struct comp_dev *dev)
+{
+	struct comp_data *kpb = comp_get_drvdata(dev);
+	uint16_t mod_id = IPC4_MOD_ID(dev_comp_id(dev));
+	uint16_t inst_id = IPC4_INST_ID(dev_comp_id(dev));
+
+	comp_dbg(dev, "kpb_unregister_ams_consumer()");
+
+	return ams_unregister_consumer(kpb->am_uuid_id, mod_id, inst_id,
+				       kpb_on_ams_kpd_notification);
+}
+#endif /* CONFIG_AMS */
```

Comment:
I don't get how this can be conditional..

Either KPB depends on AMS for functionality or it doesn't depend on AMS at all.

Something's not right here.

Reply:
@plbossart I think this means, that KPB can now use either AMS or the notifier